### PR TITLE
SF-1638 Sync based on text even when suggestions disabled

### PR DESCRIPTION
**Pull request by RaymondLuong3**
* also fix sync progress bar to show progress of based on text

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1435)

### CODE REVIEW

**Comment by Nateowami** [comment]
Testing this locally, if I have translation suggestions enabled, and I change what the project is based on, I get this error:
```
fail: EdjCase.JsonRpc.Router.Defaults.DefaultRpcInvoker[0]
      An Rpc error occurred while trying to invoke request.
      EdjCase.JsonRpc.Common.RpcException: Exception occurred from target method execution.
       ---> MongoDB.Driver.MongoWriteException: A write operation resulted in an error.
        E11000 duplicate key error collection: xforge_machine.projects index: _id_ dup key: { _id: ObjectId('62d080387d92de394d13da3a') }
       ---> MongoDB.Driver.MongoBulkWriteException`1[SIL.Machine.WebApi.Models.Project]: A bulk write operation resulted in one or more errors.
        E11000 duplicate key error collection: xforge_machine.projects index: _id_ dup key: { _id: ObjectId('62d080387d92de394d13da3a') }
         at MongoDB.Driver.MongoCollectionImpl`1.BulkWriteAsync(IClientSessionHandle session, IEnumerable`1 requests, BulkWriteOptions options, CancellationToken cancellationToken)
         at MongoDB.Driver.MongoCollectionImpl`1.UsingImplicitSessionAsync[TResult](Func`2 funcAsync, CancellationToken cancellationToken)
         at MongoDB.Driver.MongoCollectionBase`1.InsertOneAsync(TDocument document, InsertOneOptions options, Func`3 bulkWriteAsync)
         --- End of inner exception stack trace ---
         at MongoDB.Driver.MongoCollectionBase`1.InsertOneAsync(TDocument document, InsertOneOptions options, Func`3 bulkWriteAsync)
         at SIL.Machine.WebApi.DataAccess.Mongo.MongoRepository`1.InsertAsync(T entity, CancellationToken ct)
         at SIL.Machine.WebApi.Services.EngineService.AddProjectAsync(Project project)
         at SIL.XForge.Scripture.Services.SFProjectService.UpdateSettingsAsync(String curUserId, String projectId, SFProjectSettings settings) in /home/nate/src/web-xforge/src/SIL.XForge.Scripture/Services/SFProjectService.cs:line 298
         at SIL.XForge.Scripture.Controllers.SFProjectsRpcController.UpdateSettings(String projectId, SFProjectSettings settings) in /home/nate/src/web-xforge/src/SIL.XForge.Scripture/Controllers/SFProjectsRpcController.cs:line 72
         at EdjCase.JsonRpc.Router.Defaults.DefaultRpcInvoker.HandleAsyncResponses(Object returnObj)
         --- End of inner exception stack trace ---
         at EdjCase.JsonRpc.Router.Defaults.DefaultRpcInvoker.InvokeAsync(MethodInfo methodInfo, Object[] parameters, RpcRequest request, IServiceProvider serviceProvider)
         at EdjCase.JsonRpc.Router.Defaults.DefaultRpcInvoker.InvokeRequestAsync(RpcRequest request)
```
I thought perhaps the error was in some way related to #1431, even though it looks very different. But when I tried rebasing on that branch it didn't make the error go away.

**Review by RaymondLuong3 — COMMENTED** [review]
Thanks for finding this, I'll look into it

> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1435#-:-N6yTtYL3IX5fhKmsFKL:b-l6x3vq)* status: 0 of 6 files reviewed, all discussions resolved

**Review by Nateowami — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1435)* status: all files reviewed, 1 unresolved discussion (waiting on @RaymondLuong3)

___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/sync/sync-progress/sync-progress.component.spec.ts` line 143 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1435#-N6yZT2o8MsErCpg7zIc:-N6yZT2o8MsErCpg7zId:b-w25fbg) ([raw file](https://github.com/sillsdev/web-xforge/blob/e2212e45d4fb54644a0dcd2bd77efe4217f6757b/src/SIL.XForge.Scripture/ClientApp/src/app/sync/sync-progress/sync-progress.component.spec.ts#L143)):*
> ```XML
>         },
>         translateConfig: {
>           translationSuggestionsEnabled: !!args.translationSuggestionsEnabled,
> ```

My complaint (later in the file) about the use of `!!` doesn't really apply to this line, since there's a clear need here to coerce to a boolean value.
___
*[`src/SIL.XForge.Scripture/ClientApp/src/app/sync/sync-progress/sync-progress.component.spec.ts` line 165 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1435#-N6yWf1Q8ylcBtTYcCgN:-N6yWf1Q8ylcBtTYcCgO:bf211im) ([raw file](https://github.com/sillsdev/web-xforge/blob/e2212e45d4fb54644a0dcd2bd77efe4217f6757b/src/SIL.XForge.Scripture/ClientApp/src/app/sync/sync-progress/sync-progress.component.spec.ts#L165)):*
> ```XML
>         },
>         sync: {
>           queuedCount: !!args.isInProgress ? 1 : 0,
> ```

Why is `isInProgress` being coerced to boolean in order to be used as a boolean expression? That feels superfluous. The [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) has the exact same behavior.

If `(boolean | undefined) ? 1: 0` feels wrong, what about `args.isInProgress === true ? 1 : 0`. The current syntax appears to be trying to avoid the potential pitfalls of relying on which values are [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) or [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) in JavaScript, but does so by embracing them. 

In other words, in every case where `args.isInProgress ? 1 : 0` might be bad, `!!args.isInProgress ? 1 : 0` would be just as bad because it does the exact same thing.

On the other hand, maybe the `!!` is to make it clearer to the reader that the value might not be true or false, which would not be as obvious without this?

(This entire comment is fairly minor; I just find it kind of odd)

**Review by Nateowami — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1435#-:-N72SEf555-Gh-4camnN:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @RaymondLuong3)
